controllers/storagecluster/cephcluster.go

```diff
@@ -114,6 +121,24 @@ func (obj *ocsCephCluster) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.
 		return reconcile.Result{}, fmt.Errorf("'StorageDeviceSets' should not be initialized in an external CephCluster")
 	}
 
+	// The deletion function needs to run only once successfully on the cluster
+	if !deletedSucceededPodsWithDuplicateTolerations && !sc.Spec.ExternalStorage.Enable {
+		// delete the osd-prepare job completed pods
+		err = r.deleteSucceededPodsWithDuplicateTolerations(map[string]string{"app": osdPrepareLabelSelector}, sc.Namespace)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		// if cluster wide or deviceSet encryption is true, delete the osd-key-rotation cronjob completed pods
+		if sc.Spec.Encryption.ClusterWide {
```

Comment:
Don't we need to check for the other encryption as we are doing in the original PR?

Reply:
Deviceset wise encryption was not possible pre 4.18, it was only added in 4.18.

Reply:
I will add another field to this, thanks for pointing it out. Although the sc.Spec.Encryption.Enable has been deprecated since long but would be safe to add that here.

Reply:
Updated now.